phantazm-zombies/src/main/java/com/github/phantazmnetwork/zombies/equipment/gun/shoot/fire/projectile/PhantazmProjectileCollisionFilter.java

```diff
@@ -0,0 +1,53 @@
+package com.github.phantazmnetwork.zombies.equipment.gun.shoot.fire.projectile;
+
+import com.github.phantazmnetwork.commons.Namespaces;
+import com.github.phantazmnetwork.mob.MobStore;
+import com.github.steanky.ethylene.core.ConfigElement;
+import com.github.steanky.ethylene.core.collection.LinkedConfigNode;
+import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.key.Keyed;
+import net.minestom.server.entity.Entity;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+public class PhantazmProjectileCollisionFilter implements ProjectileCollisionFilter {
+
+    public record Data() implements Keyed {
+
+        public static final Key SERIAL_KEY = Key.key(Namespaces.PHANTAZM,
+                "gun.firer.projectile.collision_filter.phantazm");
+
+        @Override
+        public @NotNull Key key() {
+            return SERIAL_KEY;
+        }
+    }
+
+    public static @NotNull ConfigProcessor<Data> processor() {
+        return new ConfigProcessor<>() {
+            @Override
+            public @NotNull Data dataFromElement(@NotNull ConfigElement element) {
```

Comment:
will add a utility method to ethylene that lets you make these kinds of processors easily